libs/experimental/langchain_experimental/graph_transformers/llm.py

```diff
@@ -158,6 +159,90 @@ def map_to_base_relationship(rel: Any) -> Relationship:
     )
 
 
+def _convert_to_graph_document(
+    raw_schema: Dict,
+) -> Tuple[List[Node], Tuple[List[Relationship]]]:
+    # If there are validation errors
+    if not raw_schema["parsed"]:
+        try:
+            argument_json = json.loads(
+                raw_schema["raw"].additional_kwargs["tool_calls"][0]["function"][
+                    "arguments"
+                ]
+            )
+            nodes = []
+            for node in argument_json["nodes"]:
+                if not node.get("id"):  # Id is mandatory
+                    continue
+                nodes.append(
+                    Node(
+                        id=node["id"].title(),
+                        type=node.get("type", "Node").capitalize(),
+                    )
+                )
+            relationships = []
+            for rel in argument_json["relationships"]:
+                # Mandatory props
+                if (
+                    not rel.get("source_node_id")
+                    or not rel.get("target_node_id")
+                    or not rel.get("type")
+                ):
+                    continue
+
+                # Node type copying if needed from node list
+                if not rel.get("source_node_type"):
+                    try:
+                        rel["source_node_type"] = [
+                            el["type"]
+                            for el in argument_json["nodes"]
+                            if el["id"] == rel["source_node_id"]
+                        ][0]
+                    except IndexError:
+                        rel["source_node_type"] = "Node"
+                if not rel.get("target_node_type"):
+                    try:
+                        rel["target_node_type"] = [
+                            el["type"]
+                            for el in argument_json["nodes"]
+                            if el["id"] == rel["target_node_id"]
+                        ][0]
+                    except IndexError:
+                        rel["target_node_type"] = "Node"
+
+                source_node = Node(
+                    id=rel["source_node_id"].title(),
+                    type=rel["source_node_type"].capitalize(),
+                )
+                target_node = Node(
+                    id=rel["target_node_id"].title(),
+                    type=rel["target_node_type"].capitalize(),
+                )
+                relationships.append(
+                    Relationship(
+                        source=source_node,
+                        target=target_node,
+                        type=rel["type"].replace(" ", "_").upper(),
+                    )
+                )
+        except Exception:  # If we can't parse JSON
```

Comment:
what Exceptions could these be? shouldn't we return the partially filled node / rel-lists?

Reply:
Basically if an LLM return an incorrect JSON... I catch this error as we don't really propagate errors, but I could remove this.

If we can't parse JSON response from the LLM, there isn't much to do, except if you want to go debugging and trying to fix a JSON so that it can be parsed